drivers/axi_core/axi_dmac/axi_dmac.h

```diff
@@ -81,27 +81,40 @@ enum dma_flags {
 	DMA_LAST = 2
 };
 
+struct axi_dmac_transfer {
+	uint32_t address;
+	uint32_t size;
+};
+
 struct axi_dmac {
 	const char *name;
 	uint32_t base;
 	enum dma_direction direction;
 	uint32_t flags;
+	struct axi_dmac_transfer transfer_lifo[10];
+	volatile uint32_t driver_index;
+	volatile uint32_t user_index;
+	void (*usr_callback)();
 };
 
 struct axi_dmac_init {
 	const char *name;
 	uint32_t base;
 	enum dma_direction direction;
 	uint32_t flags;
+	void (*usr_callback)();
```

Comment:
Set the prototype as in irq.h .To be able to pass to the user a custom parameter

Reply:
This has been removed for now. Do we want callback functionality in the dma driver?